From 242b1f705af0b18233d85ad07709210a9165554e Mon Sep 17 00:00:00 2001
From: "Hebbar, Gururaja" <gururaja.hebbar@ti.com>
Date: Tue, 28 Feb 2012 15:04:50 +0530
Subject: [PATCH 0872/1092] ASoC: McASP: update resource request for DMA type

McASP device registration now uses omap_device api. This method
doesn't declare resource in order. Hence this patch replaces
platform_get_resource() with platform_get_resource_byname()

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 sound/soc/davinci/davinci-mcasp.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/sound/soc/davinci/davinci-mcasp.c b/sound/soc/davinci/davinci-mcasp.c
index d6e31e9..aaa0964 100644
--- a/sound/soc/davinci/davinci-mcasp.c
+++ b/sound/soc/davinci/davinci-mcasp.c
@@ -972,8 +972,12 @@ static int davinci_mcasp_probe(struct platform_device *pdev)
 		dma_data->dma_addr = (dma_addr_t) (pdata->tx_dma_offset +
 							mem->start);
 
-	/* first TX, then RX */
-	res = platform_get_resource(pdev, IORESOURCE_DMA, 0);
+	if (dev->version == MCASP_VERSION_3)
+		res = platform_get_resource_byname(pdev, IORESOURCE_DMA, "tx");
+	else
+		/* first TX, then RX */
+		res = platform_get_resource(pdev, IORESOURCE_DMA, 0);
+
 	if (!res) {
 		dev_err(&pdev->dev, "no DMA resource\n");
 		ret = -ENODEV;
@@ -992,7 +996,11 @@ static int davinci_mcasp_probe(struct platform_device *pdev)
 		dma_data->dma_addr = (dma_addr_t)(pdata->rx_dma_offset +
 							mem->start);
 
-	res = platform_get_resource(pdev, IORESOURCE_DMA, 1);
+	if (dev->version == MCASP_VERSION_3)
+		res = platform_get_resource_byname(pdev, IORESOURCE_DMA, "rx");
+	else
+		res = platform_get_resource(pdev, IORESOURCE_DMA, 1);
+
 	if (!res) {
 		dev_err(&pdev->dev, "no DMA resource\n");
 		ret = -ENODEV;
-- 
1.7.11.2

